### PR TITLE
Log blocks for Travis CI

### PR DESCRIPTION
**Pull request by natemcmaster**
Wraps targets in a travis logger fold. Makes log more readable. Collapses our very verbose logs on Travis into folds.

See https://travis-ci.org/aspnet/Microsoft.Data.Sqlite/builds/113208883 for example.

### CODE REVIEW

File: build/shade/_use-ci-loggers.shade [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    @{
+        bool IsTeamCity
+        {
+            get { return !string.IsNullOrEmpty(Environment.GetEnvironmentVariable("TEAMCITY_VERSION")); }
+        }
+
+        bool IsTravisCi
+        {
+            get { return !string.IsNullOrEmpty(Environment.GetEnvironmentVariable("TRAVIS")) &&
+                    !string.IsNullOrEmpty(Environment.GetEnvironmentVariable("CI")); }
```
Comment (victorhurdugaci) [inline build/shade/_use-ci-loggers.shade]: Totally unrelated: oh god, that's a terrible variable name...

File: build/shade/_use-ci-loggers.shade [hunk 2]
```diff
@@ -0,0 +1,135 @@
+functions
+    @{
+        bool IsTeamCity
```
Comment (victorhurdugaci) [inline build/shade/_use-ci-loggers.shade]: Should these go in `BuildEnv` ?

**Comment by victorhurdugaci** [comment]
Looks good assuming it doesn't break TeamCity reporting

File: build/shade/_use-ci-loggers.shade (hunk 2 above)
Comment (natemcmaster) [inline build/shade/_use-ci-loggers.shade]: Sure. Where they doesn't matter much.

File: build/shade/_use-ci-loggers.shade [hunk 3]
```diff
@@ -0,0 +1,125 @@
+functions
+    @{
+        // NB: this cannot be used along-size 'use-teamcity' because they both override StartingTarget(string)
```
Comment (BrennanConroy) [inline build/shade/_use-ci-loggers.shade]: along-side

Comment (natemcmaster) [inline build/shade/_use-ci-loggers.shade]: :stuck_out_tongue_closed_eyes: will fix

**Comment by natemcmaster** [comment]
I've tested. TeamCity logging works just fine.

File: build/shade/_use-ci-loggers.shade [hunk 4]
```diff
@@ -0,0 +1,125 @@
+functions
+    @{
+        // NB: this cannot be used along-side 'use-teamcity' because they both override StartingTarget(string)
```
Comment (dougbu) [inline build/shade/_use-ci-loggers.shade]: http://www.merriam-webster.com/dictionary/alongside (one word) :trollface:

**Comment by natemcmaster** [comment]
:heavy_check_mark: spellcheck
Any other concerns before merging?

**Comment by BrennanConroy** [comment]
I have none :+1:

**Comment by victorhurdugaci** [comment]
:shipit:

**Pull request merged**